src/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.Windows/Native/Primitives.cs

```diff
@@ -768,4 +768,33 @@ internal enum CryptAcquireFlags : int
     {
         CRYPT_ACQUIRE_ONLY_NCRYPT_KEY_FLAG = 0x00040000,
     }
+
+    [Flags]
+    internal enum ChainEngineConfigFlags : int
+    {
+        CERT_CHAIN_CACHE_END_CERT = 0x00000001,
+        CERT_CHAIN_CACHE_ONLY_URL_RETRIEVAL = 0x00000004,
+        CERT_CHAIN_USE_LOCAL_MACHINE_STORE = 0x00000008,
+        CERT_CHAIN_ENABLE_CACHE_AUTO_UPDATE = 0x00000010,
+        CERT_CHAIN_ENABLE_SHARE_STORE = 0x00000020,
+        CERT_CHAIN_DISABLE_AIA = 0x00002000,
+    }
+
+    // Windows 7 definition of the struct
+    [StructLayout(LayoutKind.Sequential)]
+    internal struct CERT_CHAIN_ENGINE_CONFIG
+    {
+        public int cbSize;
+        public IntPtr hRestrictedRoot;
+        public IntPtr hRestrictedTrust;
+        public IntPtr hRestrictedOther;
+        public int cAdditionalStore;
+        public IntPtr rghAdditionalStore;
+        public ChainEngineConfigFlags dwFlags;
+        public int dwUrlRetrievalTimeout;
+        public int MaximumCachedCertificates;
+        public int CycleDetectionModulus;
+        public IntPtr hExclusiveRoot;
+        public IntPtr hExclusiveTrustedPeople;
+    }
```

Comment:
This doesn't match the definition at https://docs.microsoft.com/en-us/windows/win32/api/wincrypt/ns-wincrypt-cert_chain_engine_config, which has an extra DWORD dwExclusiveFlags field at the end.

Reply:
The dwExclusiveFlags field is only supported by windows 8 and on. I can add 2 structs and check the os when deciding which to use.

Reply:
Since Windows 8+ understand the sizeof/versioning from Win7, I'd leave the struct as-is.